src/Compilation.zig

```diff
@@ -509,13 +509,14 @@ pub fn getSource(comp: *Compilation, id: Source.Id) Source {
 
 /// Write bytes from `reader` into `contents`, performing newline splicing,
 /// line-ending normalization (convert line endings to \n), and UTF-8 validation.
+/// `contents` will be reallocated if newline-splicing occurs
 /// Creates a Source with `contents` as the buf and adds it to the Compilation.
 /// `contents` is assumed to be large enough to hold the entire content of `reader`.
 /// `contents` must have been allocated by `comp`'s allocator since it will be reallocated
 /// if splicing occurred.
 /// Compilation owns `contents` if and only if this call succeeds; caller always retains
 /// ownership of `path`.
-pub fn addSourceFromReader(comp: *Compilation, reader: anytype, path: []const u8, contents: []u8) !Source {
+pub fn addSourceFromReader(comp: *Compilation, reader: anytype, path: []const u8, contents: *[]u8) !Source {
```

Comment:
I feel like this change could be simplified by passing in the expected size and letting `addSourceFromReader` handle allocations. Is there any use case that couldn't handle?

Reply:
I like that solution better, I've never been too satisfied with the signature of this function or the extensive doc comment to begin with. Should I do that as a separate PR against master? Not sure how long it will take to get all the allocation failure leaks ironed out.

Reply:
A separate pr would be fine.